### PR TITLE
[unitstatushandler] Send delta unit status using a dedicated struct

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: unitstatushandler/unitstatushandler.go [hunk 1]
```diff
@@ -359,13 +359,13 @@ func (instance *Instance) CloudDisconnected() {
...
 	instance.unitStatus = cloudprotocol.UnitStatus{
 		MessageType:  cloudprotocol.UnitStatusMessageType,
-		UnitConfig:   make([]cloudprotocol.UnitConfigStatus, 0),
-		Nodes:        make([]cloudprotocol.NodeInfo, 0),
-		Services:     make([]cloudprotocol.ServiceStatus, 0),
-		Instances:    make([]cloudprotocol.InstanceStatus, 0),
-		Layers:       make([]cloudprotocol.LayerStatus, 0),
-		Components:   make([]cloudprotocol.ComponentStatus, 0),
-		UnitSubjects: make([]string, 0),
+		UnitConfig:   nil,
```
Comment (al1img) [inline unitstatushandler/unitstatushandler.go:362]: It means that we will omit these fields even for full unit status? Looks like it is not expected behavior

Comment (mlohvynenko) [reply unitstatushandler/unitstatushandler.go:362]: As per public docs these fields have default value null.
I didn't find that null is applicable only if delta unit status is true.

So my assumption that should send json tags only if they have non-default values set (not null)

File: vendor/github.com/aosedge/aos_common/api/cloudprotocol/unitstatus.go [hunk 2]
```diff
@@ -171,13 +171,13 @@ type ComponentStatus struct {
 type UnitStatus struct {
 	MessageType  string             `json:"messageType"`
 	IsDeltaInfo  bool               `json:"isDeltaInfo"`
-	UnitConfig   []UnitConfigStatus `json:"unitConfig"`
-	Nodes        []NodeInfo         `json:"nodes"`
-	Services     []ServiceStatus    `json:"services"`
-	Instances    []InstanceStatus   `json:"instances"`
+	UnitConfig   []UnitConfigStatus `json:"unitConfig,omitempty"`
```
Comment (mykola-kobets-epam) [inline vendor/github.com/aosedge/aos_common/api/cloudprotocol/unitstatus.go:174]: does it mean that we don't provide empty array at all to the cloud or it omits a nil value only?

File: go.mod [hunk 3]
```diff
@@ -6,6 +6,8 @@ replace github.com/ThalesIgnite/crypto11 => github.com/aosedge/crypto11 v1.0.3-0
 
 replace github.com/anexia-it/fsquota => github.com/aosedge/fsquota v0.0.0-20231127111317-842d831105a7
 
+replace github.com/aosedge/aos_common => ../aos_common
```
Comment (al1img) [inline go.mod:9]: Do not forget to remove it.

File: unitstatushandler/unitstatushandler_internal_test.go [hunk 4]
```diff
@@ -1733,11 +1733,25 @@ func NewTestSender() (sender *TestSender) {
 }
 
 func (sender *TestSender) SendUnitStatus(unitStatus cloudprotocol.UnitStatus) (err error) {
+	if json, err := json.Marshal(unitStatus); err == nil {
```
Comment (al1img) [inline unitstatushandler/unitstatushandler_internal_test.go:1736]: Remove this debug log

File: unitstatushandler/unitstatushandler_internal_test.go [hunk 5]
```diff
@@ -1733,11 +1733,25 @@ func NewTestSender() (sender *TestSender) {
...
+		log.Debugf("Send unit status: %s", string(json))
+	}
+
 	sender.statusChannel <- unitStatus
 
 	return nil
 }
 
+func (sender *TestSender) SendDeltaUnitStatus(deltaUnitStatus cloudprotocol.DeltaUnitStatus) (err error) {
+	if json, err := json.Marshal(deltaUnitStatus); err == nil {
```
Comment (al1img) [inline unitstatushandler/unitstatushandler_internal_test.go:1746]: ditto

**Review by al1img — APPROVED** [review]

**Pull request merged**